recipes/onnxruntime/all/patches/1.15.1-0001-cmake-dependencies.patch

```diff
@@ -0,0 +1,172 @@
+ if (NOT onnxruntime_USE_FULL_PROTOBUF)
+-  target_compile_definitions(onnx PUBLIC "__ONNX_NO_DOC_STRINGS")
++  # target_compile_definitions(onnx PUBLIC "__ONNX_NO_DOC_STRINGS")
+ endif()
+ 
+ if (onnxruntime_RUN_ONNX_TESTS)
+diff --git a/cmake/external/wil.cmake b/cmake/external/wil.cmake
+index d38535c4a1..02d6dbd85d 100644
+--- a/cmake/external/wil.cmake
++++ b/cmake/external/wil.cmake
+@@ -9,14 +9,10 @@ FetchContent_Declare(
+ 	URL_HASH SHA1=${DEP_SHA1_microsoft_wil}
+     FIND_PACKAGE_ARGS NAMES wil
+ )
+-#We can not use FetchContent_MakeAvailable(microsoft_wil) at here, since their cmake file
+-#always executes install command without conditions.
+-FetchContent_Populate(microsoft_wil)
+-if(NOT wil_FOUND)
++
++if(WIN32)
++  FetchContent_MakeAvailable(microsoft_wil)
++else()
+   add_library(WIL INTERFACE)
+   add_library(WIL::WIL ALIAS WIL)
+-
+-  # The interface's include directory.
+-  target_include_directories(WIL INTERFACE
+-    $<BUILD_INTERFACE:${microsoft_wil_SOURCE_DIR}/include>)
+-endif()
+\ No newline at end of file
++endif()
+diff --git a/cmake/external/xnnpack.cmake b/cmake/external/xnnpack.cmake
+index 1fc2c6ccdc..a2c28957f9 100644
+--- a/cmake/external/xnnpack.cmake
++++ b/cmake/external/xnnpack.cmake
+@@ -25,12 +25,15 @@ set(FXDIV_SOURCE_DIR ${fxdiv_SOURCE_DIR})
+ 
+ FetchContent_Declare(pthreadpool URL ${DEP_URL_pthreadpool} URL_HASH SHA1=${DEP_SHA1_pthreadpool})
+ onnxruntime_fetchcontent_makeavailable(pthreadpool)
+-FetchContent_Declare(googlexnnpack URL ${DEP_URL_googlexnnpack}  URL_HASH SHA1=${DEP_SHA1_googlexnnpack}
+-PATCH_COMMAND ${Patch_EXECUTABLE} --binary --ignore-whitespace -p1 < ${PROJECT_SOURCE_DIR}/patches/xnnpack/AddEmscriptenAndIosSupport.patch)
++FetchContent_Declare(
++    googlexnnpack
++    URL ${DEP_URL_googlexnnpack}
++    URL_HASH SHA1=${DEP_SHA1_googlexnnpack}
++    PATCH_COMMAND ${Patch_EXECUTABLE} --binary --ignore-whitespace -p1 < ${PROJECT_SOURCE_DIR}/patches/xnnpack/AddEmscriptenAndIosSupport.patch
++    FIND_PACKAGE_ARGS NAMES xnnpack
++)
+ 
+ onnxruntime_fetchcontent_makeavailable(googlexnnpack)
+-set(XNNPACK_DIR ${googlexnnpack_SOURCE_DIR})
+-set(XNNPACK_INCLUDE_DIR ${XNNPACK_DIR}/include)
+ 
+ set(onnxruntime_EXTERNAL_LIBRARIES_XNNPACK XNNPACK pthreadpool)
+ 
+diff --git a/cmake/onnxruntime_common.cmake b/cmake/onnxruntime_common.cmake
+index 0410d3361c..685df7a487 100644
+--- a/cmake/onnxruntime_common.cmake
++++ b/cmake/onnxruntime_common.cmake
+@@ -195,7 +195,7 @@ if (ARM64 OR ARM OR X86 OR X64 OR X86_64)
+     # Its functionality in detecting x86 cpu features are lacking, so is support for Windows.
+     if (CPUINFO_SUPPORTED)
+       onnxruntime_add_include_to_target(onnxruntime_common cpuinfo::cpuinfo)
+-      list(APPEND onnxruntime_EXTERNAL_LIBRARIES cpuinfo::cpuinfo cpuinfo::clog)
++      list(APPEND onnxruntime_EXTERNAL_LIBRARIES cpuinfo::cpuinfo)
```

Comment:
I don't think this is 100% correct. From cpuinfo CMakeLists.txt it's implied that both `cpuinfo::cpuinfo` and `cpuinfo::clog` targets are exported: https://github.com/pytorch/cpuinfo/blob/082deffc80ce517f81dc2f3aebe6ba671fcd09c9/CMakeLists.txt#L285 but conan cpuinfo recipe doesn't define components (when it should).

Reply:
Seems the situation with`cpuinfo::clog` is way cumbersome. It is exposed as a aliased target since [Jun 18, 2022](https://github.com/pytorch/cpuinfo/commit/082deffc80ce517f81dc2f3aebe6ba671fcd09c9) and removed altogether in [Jan 18, 2023](https://github.com/pytorch/cpuinfo/commit/3dc310302210c1891ffcfb12ae67b11a3ad3a150).

Reply:
https://github.com/conan-io/conan-center-index/pull/18164